RHDEVDOCS-4431 - 5.4.7, 5.5.3 - Logging Z-Stream Release Notes (#51478)
openshift/openshift-docs: modules/cluster-logging-rn-5.5.3.adoc

//module included in cluster-logging-release-notes.adoc
:_content-type: REFERENCE
[id="cluster-logging-release-notes-5-5-3_{context}"]
= Logging 5.5.3
This release includes link:https://access.redhat.com/errata/RHBA-2022:6858[OpenShift Logging Bug Fix Release 5.5.3].

[id="openshift-logging-5-5-3-bug-fixes_{context}"]
== Bug fixes

> * Before this update log entries that had structured messages included the original message field which made the entry larger. This update removes the message field for structured logs to reduce the increased size. (link:https://issues.redhat.com/browse/LOG-2759[LOG-2759])

Reviewer: Missing comma:

`Before this update, log entries that had structured messages...`

> * Before this update, log entries that had structured messages included the original message field which made the entry larger. This update removes the message field for structured logs to reduce the increased size. (link:https://issues.redhat.com/browse/LOG-2759[LOG-2759])

Reviewer 2: ```suggestion
* Before this update, log entries that had structured messages included the original message field, which made the entry larger. This update removes the message field for structured logs to reduce the increased size. (link:https://issues.redhat.com/browse/LOG-2759[LOG-2759])
```

* Before this update, the collector configuration excluded logs from `collector`, `default-log-store`, and `visualization`  pods, but was unable to exclude logs archived in a `.gz` file. With this update, archived logs stored as `.gz` files of `collector`, `default-log-store`, and `visualization` pods are also excluded. (link:https://issues.redhat.com/browse/LOG-2844[LOG-2844])

> * Before this update, when requests to an unavailable pod was sent through the gateway, no alert would fire to warn of the disruption. With this update, the individual alerts will fire if the gateway has issues completing a write or read request respectively. (link:https://issues.redhat.com/browse/LOG-2884[LOG-2884])

Reviewer: `fire` might be too colloquial of a verb. 

also, verb-tense agreement with `requests` and `was` 

what do you think of this rephrase?

`Before this update, when requests to an unavailable pod were sent through the gateway, no alert would warn of the disruption.`

> * Before this update, pod metadata could be altered by fluent plugins because the values passed through the pipeline by reference. This update fixes that to ensure each log message receives a copy of the pod metadata so each message processes independently. (link:https://issues.redhat.com/browse/LOG-3046[LOG-3046])

Reviewer: I'm not sure what `fixes that` refers to. Does `that` refer to the pod metadata, the values, the pipeline references? 

What do you think of something like this?

`This update fixes how pod metadata is referenced by ensuring each log message receives a copy of the pod metadata, so now each message independently processes its reference.` 

Note, I wasn't sure what the messages are processing. I inferred that it was processing its reference. I might be incorrect.

Author: "This update ensures each log message receives a copy of the pod metadata so each message processes independently."

> * Before this update, selecting 'unknown' severity in the OpenShift Console Logs view excluded logs with a `level=unknown` value. With this update, logs without level and with `level=unknown` values are visible when filtering by 'unknown' severity. (link:https://issues.redhat.com/browse/LOG-3062[LOG-3062])

Reviewer 2: ```suggestion
* Before this update, selecting *unknown* severity in the OpenShift Console Logs view excluded logs with a `level=unknown` value. With this update, logs without level and with `level=unknown` values are visible when filtering by *unknown* severity. (link:https://issues.redhat.com/browse/LOG-3062[LOG-3062])
```
If `unknown` is a UI element, it should be in bold type. If it's a literal, it should be in backticks. I'm not sure which it should be.

> * Before this update, log records sent to Elasticsearch had an extra field named `write-index` which contains the name of the index to which the logs need to be sent. This field is not a part of the data model. After this update, this field is no longer sent. (link:https://issues.redhat.com/browse/LOG-3075[LOG-3075])

Reviewer 2: ```suggestion
* Before this update, log records sent to Elasticsearch had an extra field named `write-index` that contained the name of the index to which the logs need to be sent. This field is not a part of the data model. After this update, this field is no longer sent. (link:https://issues.redhat.com/browse/LOG-3075[LOG-3075])
```
Past tense for the "before this update" sentence.

> * With the introduction of the new built-in link:https://cloud.redhat.com/blog/pod-security-admission-in-openshift-4.11[Pod Security Admission Controller]Pods that not configured in accordance with the enforced security standards defined globally or on the namespace level cannot run. With this update, the Operator and collectors allow privileged execution and will run without security audit warnings or errors. (link:https://issues.redhat.com/browse/LOG-3077[LOG-3077])

Reviewer 2: ```suggestion
* With the introduction of the new built-in link:https://cloud.redhat.com/blog/pod-security-admission-in-openshift-4.11[Pod Security Admission Controller], Pods not configured in accordance with the enforced security standards defined globally or on the namespace level cannot run. With this update, the Operator and collectors allow privileged execution and run without security audit warnings or errors. (link:https://issues.redhat.com/browse/LOG-3077[LOG-3077])
```
Fix typos and grammar.

> * With the introduction of the new built-in link:https://cloud.redhat.com/blog/pod-security-admission-in-openshift-4.11[Pod Security Admission Controller] that enforces the Pod Security Standards, Pods that not configured in accordance with the enforced security standards defined globally or on the namespace level cannot run. With this update, the operator and collectors allow privileged execution and will run without security audit warnings or errors. (link:https://issues.redhat.com/browse/LOG-3077[LOG-3077])

Reviewer: I had to read this a couple of times to understand it. Let me know if I am understanding it correctly and what you think of this rephrase:

`With the introduction of the new built-in link:https://cloud.redhat.com/blog/pod-security-admission-in-openshift-4.11[Pod Security Admission Controller], it enforces the Pod Security Standards. Security standards provide global definitions and namespace level definitions. Pods that are not configured in accordance with either of these definitions cannot run. With this update, the operator and collectors allow privileged execution and run without security audit warnings or errors.`

> * Before this update, the Operator removed any custom outputs defined in the `ClusterLogForwarder` custom resource when using LokiStack as the default log storage. With this update, the Operator merges custom outputs with the default outputs when processing the `ClusterLogForwarder` custom resource. (link:https://issues.redhat.com/browse/LOG-3095[LOG-3095])

Reviewer: In your previous bullet point, it was `operator` now in this bullet point, it's `Operator` 

Is it a proper noun and therefore should be capitalized? 

It doesn't read as a proper noun to me, so I think it can be lowercase. 

I defer to you, as long as it's consistent.

Author: Operator is always capitalized per the [OpenShift docs guidelines](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#operator-name-capitalization), thanks for catching that!
